### ISSUE TITLE
Weird first visualization

### ISSUE THREAD

**Issue by felixveysseyre** [Kitware/paraviewweb#319]
Hello,

Thanks to @jourdain , I was able to build my one visualizer base on ParaViewWeb npm package.
When used on a local computer (both Ubuntu and MacOS), everything is working smootly.

I tried to deploy my project on a g2.2xlarge AWS instance to test performances in real condition.
I had some hard time finding how to install X and nvidia drivers but thanks to this [issue](http://askubuntu.com/questions/429596/how-to-choose-the-vga-when-setting-up-the-x-server) I finally succeed.

I tested my project and it seems to works correctly, nevertheless I always get some weird results for my very first connection (after I started the launcher).

Here are two screenshots representing what I got for my first connection and for any other connection:
![capture d ecran 2017-01-12 a 16 04 08](https://cloud.githubusercontent.com/assets/4909259/21895773/2f731c78-d8e4-11e6-94b8-060c77e4f596.png)
![capture d ecran 2017-01-12 a 16 04 16](https://cloud.githubusercontent.com/assets/4909259/21895774/2f75a3d0-d8e4-11e6-990c-75a457ed4a22.png)

I have looked into ParaViewWeb logs and nothing special appeared.

Here are some test cases:

Test1:
 - Start launcher
 - Start new connection: connection 1 -> weird
 - Start new connection: connection 2 -> OK
 - Start new connection: connection 3 -> OK

Test2:
 - Start launcher
 - Start new connection: connection 1 -> weird
 - Close connection 1
 - Start new connection: connection 2 -> OK
 - Start new connection: connection 3 -> OK

Any idea guys ?

Thanks for your time !

**Comment by jourdain** [comment]
And interacting with the view does not fix the rendering issue?

**Comment by jourdain** [comment]
Which window manager did you install on the EC2 instance?

You may want to use offscreen rendering to see if that fix the issue. 

Just need to add similar lines to your server 135+136

https://github.com/Kitware/light-viz/blob/master/server/light_viz_protocols.py#L135

**Comment by jourdain** [comment]
Moreover, did you picked the latest driver or did you simply use the link that was provided. Moreover did you double check the actual GPU?

**Comment by felixveysseyre** [comment]
Interacting with the view changes the view but it does not fix the issue:

![capture d ecran 2017-01-12 a 17 08 59](https://cloud.githubusercontent.com/assets/4909259/21897446/de3d1bc8-d8e9-11e6-80e7-0103e1d1dd09.png)

I changed my code to render the view offscreen. The rendering is now done offscreen but the issue is still here.

When setting-up my EC2 instance, I installed the `xserver-xorg` package. Therefore I used X as window manager.

**Comment by felixveysseyre** [comment]
When setting-up my EC2 instance, I installed 340.46 nvidia drivers.

I tried more recent drivers (346 & 364) on other instances and I did get errors preventing me to use OpenGL correctly.

What do you advise to double check the GPU ?

**Comment by jourdain** [comment]
Don't know the command to show the hardware information.

Is any of the command line provided [here](http://kitware.github.io/paraviewweb/docs/nvidia_redhat.html) provide that information in output?

**Comment by felixveysseyre** [comment]
The GPU available on my EC2 instance is `GK104GL [GRID K520]`

```
00:02.0 VGA compatible controller [0300]: Cirrus Logic GD 5446 [1013:00b8]
00:03.0 VGA compatible controller [0300]: NVIDIA Corporation GK104GL [GRID K520] [10de:118a] (rev a1)
```

**Comment by jourdain** [comment]
[this](http://kitware.github.io/paraviewweb/docs/ansible.html) suggest the UI to make sure you download the right driver.

**Comment by felixveysseyre** [comment]
I did installed latest Nvidia drivers available (367.57) and it did solve the issue.

Nevertheless I think I found the root of the issue.

I was starting X in a bad way:
```
sudo /usr/bin/X &
``` 

He was logging messages anywhere:

```
7 XSELINUXs still allocated at reset
SCREEN: 0 objects of 368 bytes = 0 total bytes 0 private allocs
COLORMAP: 0 objects of 8 bytes = 0 total bytes 0 private allocs
DEVICE: 0 objects of 96 bytes = 0 total bytes 0 private allocs
CLIENT: 0 objects of 128 bytes = 0 total bytes 0 private allocs
WINDOW: 0 objects of 88 bytes = 0 total bytes 0 private allocs
PIXMAP: 2 objects of 120 bytes = 240 total bytes 0 private allocs
...
```

Rather than starting X that way, I am now using `xinit`:
```
sudo xinit
```

Hopefully it will help some one !
